### PR TITLE
Made the code handle errors

### PR DESCRIPTION
**Pull request by coderwithsense**
I have just changed the way of writing "ingest.py" and "privateGPT.py" files
Now they handle errors and are in more readable manner

### CODE REVIEW

File: privateGPT.py [hunk 1]
```diff
@@ -4,36 +4,45 @@
...
+            print(query)
+            print("\n> Answer:")
+            print(answer)
+
+            # Print the relevant sources used for the answer
+            for document in docs:
+                print("\n> " + document.metadata["source"] + ":")
+                print(document.page_content)
+
+    except FileNotFoundError:
```
Comment (imartinez) [inline privateGPT.py:41]: Not sure this FileNotFoundError is relevant here, given there is no manual file input by the user. I'd remove it.

**Review by imartinez — COMMENTED** [review]
Thanks for the improvements! Definitely worth merging. Please take a look at my only comment.

File: privateGPT.py (hunk 1 above)
Comment (coderwithsense) [reply privateGPT.py:41]: Surely, i will update that too

**Comment by imartinez** [comment]
@coderwithsense please update the branch with latest changes on main so it can be merged. Shouldn't be hard to integrate. Thanks!

**Comment by rabotone** [comment]
How to deploy

File: privateGPT.py (hunk 1 above)
Comment (coderwithsense) [reply privateGPT.py:41]: okay i will do the same

**Comment by coderwithsense** [comment]
> @coderwithsense please update the branch with latest changes on main so it can be merged. Shouldn't be hard to integrate. Thanks!

okayy

File: ingest.py [hunk 2]
```diff
@@ -1,22 +1,41 @@
+import sys
 from langchain.document_loaders import TextLoader
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 from langchain.vectorstores import Chroma
 from langchain.embeddings import LlamaCppEmbeddings
```
Comment (jayresolf132) [inline ingest.py:5]: Sorry.

File: privateGPT.py (hunk 1 above)
Comment (jayresolf132) [reply privateGPT.py:41]: > Not sure this FileNotFoundError is relevant here, given there is no manual file input by the user. I'd remove it.

Yo!

File: ingest.py [hunk 3]
```diff
@@ -1,22 +1,41 @@
...
+    if len(sys.argv) < 2:
+        print("Please provide the path to the input file as a command line argument.")
+        return
+
+    input_file_path = sys.argv[1]
+
+    try:
+        # Load document and split into chunks
+        loader = TextLoader(input_file_path, encoding="utf8")
+        documents = loader.load()
```
Comment (jayresolf132) [inline ingest.py:18]: Nay!

**Pull request closed without merging**